test/test_python_dispatch.py

```diff
@@ -79,6 +80,117 @@ def my_sum(*args, **kwargs):
+
+            self.assertEqual(torch.where(cond, x, y), torch.tensor([-1, -2, 3]))
+            self.assertTrue(CALLED[0])
+            del my_lib
+
+        def override_gelu_cuda() -> None:
+            # Example 2: Use relu to approximate gelu for faster compute
+            fastest_gelu_code_string = '''
+            template <typename T> T fast_gelu(T a){
+                return a > 0 ? a : 0;
+            }
+            '''
+            jitted_gelu = _create_jit_fn(fastest_gelu_code_string)
+
+            CALLED = [False]
+
+            def fast_gelu(*args, **kwargs):
+                CALLED[0] = True
+                return jitted_gelu(*args, **kwargs)
+
+            # overriding gelu's cuda kernel with Jiterator generated relu kernel
+            my_lib = Library("aten", "IMPL")
+            my_lib.impl('aten::gelu', fast_gelu, "CUDA")
+
+            x = torch.rand([3, 3], device='cuda', dtype=torch.float)
+            self.assertEqual(torch.nn.functional.gelu(x), torch.nn.functional.relu(x))
+            self.assertTrue(CALLED[0])
+            del my_lib
+
+        def override_exp_cuda() -> None:
+            # Example 3: Preventing exp from exploding for float16
+            clipped_exp_code_string = '''
+            template <typename T> T clipped_exp(T a){
+                return a > T(10.0) ? T(22026.4657948) : exp(a);
+            }
+            '''
+            jitted_exp = _create_jit_fn(clipped_exp_code_string)
+
+            CALLED = [False]
+
+            def clipped_exp(*args, **kwargs):
+                CALLED[0] = True
+                return jitted_exp(*args, **kwargs)
+
+            # overriding exp's cuda kernel with clipped_exp kernel
+            my_lib = Library("aten", "IMPL")
+            my_lib.impl('aten::exp', clipped_exp, "CUDA")
+
+            x = torch.tensor([0.0, 100.0], device='cuda', dtype=torch.float16)
+            self.assertEqual(torch.exp(x), torch.tensor([1.0, 22026.4657948], dtype=torch.float16))
+            self.assertTrue(CALLED[0])
+            del my_lib
+
+        def override_add_cuda() -> None:
+            # Example 4: simulate a hardware bug, where the adder is always off by 1
+            buggy_add_code_string = '''
+            template <typename T> T buggy_add(T a, T b){
+                return a + b + T(1);
+            }
+            '''
+            jitted_add = _create_jit_fn(buggy_add_code_string)
+
+            CALLED = [False]
+
+            def buggy_add(*args, **kwargs):
+                CALLED[0] = True
+                return jitted_add(*args, **kwargs)
+
+            my_lib = Library("aten", "IMPL")
+            my_lib.impl('aten::add.Tensor', buggy_add, "CUDA")
+
+            x = torch.rand([3, 3], device='cpu')
+            y = torch.rand([3], device='cpu')
+
+            x_cuda = x.cuda()
+            y_cuda = y.cuda()
+
+            self.assertEqual(x_cuda + y_cuda, x + y + 1)
+            self.assertTrue(CALLED[0])
+            del my_lib
```

Comment:
add a check that the old behavior is retained after your library is deleted

Reply:
done.